tests/ops/fixtures/saas/doordash_fixtures.py

```diff
@@ -0,0 +1,181 @@
+def doordash_dataset() -> Dict[str, Any]:
+    return load_dataset_with_replacement(
+        "data/saas/dataset/doordash_dataset.yml",
+        "<instance_fides_key>",
+        "doordash_instance",
+    )[0]
+
+
+@pytest.fixture(scope="function")
+def doordash_connection_config(
+    db: session, doordash_config, doordash_secrets
+) -> Generator:
+    fides_key = doordash_config["fides_key"]
+    connection_config = ConnectionConfig.create(
+        db=db,
+        data={
+            "key": fides_key,
+            "name": fides_key,
+            "connection_type": ConnectionType.saas,
+            "access": AccessLevel.write,
+            "secrets": doordash_secrets,
+            "saas_config": doordash_config,
+        },
+    )
+    yield connection_config
+    connection_config.delete(db)
+
+
+@pytest.fixture
+def doordash_dataset_config(
+    db: Session,
+    doordash_connection_config: ConnectionConfig,
+    doordash_dataset: Dict[str, Any],
+) -> Generator:
+    fides_key = doordash_dataset["fides_key"]
+    doordash_connection_config.name = fides_key
+    doordash_connection_config.key = fides_key
+    doordash_connection_config.save(db=db)
+    dataset = DatasetConfig.create(
+        db=db,
+        data={
+            "connection_config_id": doordash_connection_config.id,
+            "fides_key": fides_key,
+            "dataset": doordash_dataset,
+        },
+    )
+    yield dataset
+    dataset.delete(db=db)
+
+
+@pytest.fixture()
+def doordash_postgres_dataset() -> Dict[str, Any]:
+    return {
+        "fides_key": "doordash_postgres",
+        "name": "Doordash Postgres",
+        "description": "Lookup for Doordash delivery IDs",
+        "collections": [
+            {
+                "name": "doordash_deliveries",
+                "fields": [
+                    {
+                        "name": "email",
+                        "data_categories": ["user.contact.email"],
+                        "fidesops_meta": {"data_type": "string", "identity": "email"},
+                    },
+                    {
+                        "name": "delivery_id",
+                        "fidesops_meta": {"data_type": "string"},
+                    },
+                ],
+            }
+        ],
+    }
+
+
+@pytest.fixture
+def doordash_postgres_dataset_config(
+    connection_config: ConnectionConfig,
+    doordash_postgres_dataset: Dict[str, Any],
+    db: Session,
+) -> Generator:
+    fides_key = doordash_postgres_dataset["fides_key"]
+    connection_config.name = fides_key
+    connection_config.key = fides_key
+    connection_config.save(db=db)
+    dataset = DatasetConfig.create(
+        db=db,
+        data={
+            "connection_config_id": connection_config.id,
+            "fides_key": fides_key,
+            "dataset": doordash_postgres_dataset,
+        },
+    )
+    yield dataset
+    dataset.delete(db=db)
+
+
+@pytest.fixture(scope="function")
+def doordash_postgres_db(postgres_integration_session):
```

Comment:
this is a nice approach for getting a local env setup for external dataset reference testing.

since we anticipate this sort of setup being needed for more saas connectors (i.e. those that require external dataset references), i think we should try to DRY it up as much as possible and extract this into a shared util function. i already see numerous places in the codebase repeating the same logic. i'll give a quick shot implementing that shared util